models/session.js

```diff
@@ -0,0 +1,30 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const PlaySessionSchema = new Schema({
+  userId: {
+    type: String,
+    required: true,
+  },
+  sessionId: {
+    type: String,
+    requires: true,
+  },
+  difficulty: {
+    type: Number,
+    required: true,
+  },
+  length: {
+    type: Number,
+    required: true,
+  },
+  sessionData: {
+    type: Array,
```

Comment:
Nice schema here, although for session data I think it would be nice to define nested values as well.

Reply:
could you perhaps hint on the correct syntax here, am struggling to find how to define the nested elements here

Reply:
I think you can write it in standard object notation pattern but you could use create a child Schema. Anyways, refer to this section of mongoose docs - https://mongoosejs.com/docs/subdocs.html#subdocuments-versus-nested-paths